.docker/ci/Dockerfile

```diff
@@ -1,50 +1,82 @@
-# moveit/moveit:melodic-ci
-# Sets up a base image to use for running Continuous Integration on Travis
-
-FROM ros:melodic-ros-base
-MAINTAINER Dave Coleman dave@picknik.ai
-
-ENV TERM xterm
-
-# Setup catkin workspace
-ENV CATKIN_WS=/root/ws_moveit
-WORKDIR $CATKIN_WS
-
-# Commands are combined in single RUN statement with "apt/lists" folder removal to reduce image size
-# https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers
-RUN \
-    mkdir src && \
-    cd src && \
-    #
-    # Download moveit source, so that we can get necessary dependencies
-    wstool init --shallow . https://raw.githubusercontent.com/ros-planning/moveit/${ROS_DISTRO}-devel/moveit.rosinstall && \
-    #
-    # Update apt package list as previous containers clear the cache
-    apt-get -qq update && \
-    apt-get -qq dist-upgrade && \
-    #
-    # Install some base dependencies
-    apt-get -qq install -y \
-        # Some source builds require a package.xml be downloaded via wget from an external location
-        wget \
-        # Required for rosdep command
-        sudo \
-        # Preferred build tools
-        python-catkin-tools \
-        clang clang-format-3.9 clang-tidy clang-tools \
-        ccache && \
-    #
-    # Download all dependencies of MoveIt!
-    rosdep update && \
-    rosdep install -y --from-paths . --ignore-src --rosdistro ${ROS_DISTRO} --as-root=apt:false && \
-    #
-    # Remove the source code from this container
-    # TODO: in the future we may want to keep this here for further optimization of later containers
-    cd .. && \
-    rm -rf src/ && \
-    #
-    # Clear apt-cache to reduce image size
-    rm -rf /var/lib/apt/lists/*
-
-# Continous Integration Setting
-ENV IN_DOCKER 1
+# This dockerfile expects proxies to be set via --build-arg if needed
+# It also expects to be contained in the /moveit2 root folder for file copy
+# Example build command:
+# export http_proxy=http://my.proxy.com:80
+# export CMAKE_BUILD_TYPE=Debug
+# docker build -t nav2:latest --build-arg http_proxy --build-arg CMAKE_BUILD_TYPE ./
+FROM osrf/ros2:nightly
+
+# setup keys
+ARG http_proxy
+RUN if [ "$http_proxy" != "" ]; \
+    then \
+      apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 \
+      --keyserver-options http-proxy=$http_proxy \
+      --recv-keys 421C365BD9FF1F717815A3895523BAEEB01FA116; \
+    fi
+
+# install ROS2 dependencies
+RUN apt-get update && apt-get install -q -y \
+      build-essential \
+      cmake \
+      git \
+      python3-colcon-common-extensions \
+      python3-vcstool \
+      wget \
+    && rm -rf /var/lib/apt/lists/*
+
+# copy ros package repo
+ENV MOVEIT2_WS /opt/moveit2_ws
+RUN mkdir -p $MOVEIT2_WS/src
```

Comment:
I don't think we should be putting the workspace in `/opt`. Isn't that generally reserved for system installs? In the past we've been putting the build in something like: `ENV ROS_WS=/root/ws_moveit`

Reply:
I original helped with those dockerfiles as assuming the convention for root, and I think I was mistaken to do so. This makes it tricky to use the same workspace with changing to a different users, e.g. using your host machine user mounted in the container with x forwarding when testing GUIs etc,  

https://github.com/osrf/docker_images/pull/243
https://github.com/ros-planning/moveit2/pull/46#discussion_r267151753

Reply:
sounds good